chemprop/models/model.py

```diff
@@ -212,21 +217,44 @@ def predict_step(self, batch: TrainingBatch, batch_idx: int, dataloader_idx: int
         return self(bmg, X_vd, X_d)
 
     def configure_optimizers(self):
-        opt = optim.Adam(self.parameters(), self.init_lr)
-
-        lr_sched = NoamLR(
-            opt,
-            self.warmup_epochs,
-            self.trainer.max_epochs,
-            self.trainer.estimated_stepping_batches // self.trainer.max_epochs,
-            self.init_lr,
-            self.max_lr,
-            self.final_lr,
+        no_decay = ["bias", "bn.weight"]
+        optimizer_grouped_parameters = [
+            {
+                "params": [
+                    p for n, p in self.named_parameters() if not any(nd in n for nd in no_decay)
+                ],
+                "weight_decay": self.weight_decay,
+            },
+            {
+                "params": [
+                    p for n, p in self.named_parameters() if any(nd in n for nd in no_decay)
+                ],
+                "weight_decay": 0.0,
+            },
+        ]
+
+        opt = optim.AdamW(optimizer_grouped_parameters, self.init_lr)
+        steps_per_epoch = self.trainer.num_training_batches
+        warmup_steps = self.warmup_epochs * steps_per_epoch
+        if self.trainer.max_epochs == -1:
+            warnings.warn(
+                "For infinite training, the number of cooldown epochs in learninig rate scheduler is set to 100 times the number of warmup epochs."
+            )
+            cooldown_steps = 100 * warmup_steps
+        else:
+            cooldown_epochs = self.trainer.max_epochs - self.warmup_epochs
+            cooldown_steps = cooldown_epochs * steps_per_epoch
+
+        lr_kwargs = dict(
+            warmup_steps=warmup_steps,
+            cooldown_steps=cooldown_steps,
+            init_lr=self.init_lr,
+            max_lr=self.max_lr,
+            final_lr=self.final_lr,
         )
-        lr_sched_config = {
-            "scheduler": lr_sched,
-            "interval": "step" if isinstance(lr_sched, NoamLR) else "batch",
-        }
+        lr_sched = NoamLR(opt, **lr_kwargs)
```

Comment:
why group into a dictionary then unpack in the very next line?

Reply:
I originally designed this for multiple learning rate schedulers, but I opted to use only the current learning rate scheduler, so I should just pass it directly.